src/guiapi/src/Rect16.cpp

```diff
@@ -150,37 +150,66 @@ void Rect16::Align(Rect16 rc, uint8_t align) {
     }
 }
 
-void Rect16::VerticalSplit(Rect16 splits[], Rect16 spaces[], size_t count, uint16_t spacing) const {
+void Rect16::HorizontalSplit(Rect16 splits[], Rect16 spaces[], size_t count, uint16_t spacing, uint8_t ratio[]) const {
     if (count == 0)
         return;
     if (count == 1) {
         splits[0] = *this;
         return;
     }
 
-    uint16_t width = Width() / count - spacing * (count - 1);
-
-    Rect16 rc({ 0, Top() }, width, Height());
-    Rect16 rc_space({ 0, Top() }, spacing, Height());
-    size_t index;
-    size_t right = BottomRight().x - width;
-
-    for (index = 0; index < count / 2; ++index) {
-        splits[index] = rc + Rect16::Left_t(Left() + index * (width + spacing));            // 1 from begin
-        splits[count - 1 - index] = rc + Rect16::Left_t(right - index * (width + spacing)); // 1 from end
+    size_t index, fullRatio = 0;
+    uint16_t width = (Width() - (spacing * (count - 1))) / count;
+    float r;
+    if (ratio != nullptr) {
+        for (index = 0; index < count; index++) {
+            fullRatio += ratio[index];
+        }
+    }
+    for (index = 0; index < count; index++) {
+        if (ratio != nullptr) {
+            r = (float)ratio[index] / (float)fullRatio;
+            width = Width() - (spacing * (count - 1));
```

Comment:
`Width() - (spacing * (count - 1))` is const so it can be pre-computed (as a const)